src/main/java/service/OptionService.java

```diff
@@ -1,70 +1,49 @@
 package service;
 
 
-import cache.CarCache;
+
 import cache.OptionCache;
 import dao.*;
-//import dao.OptionDAO;
-import data.Car;
 import data.CarOption;
 import data.Option;
-
+import enums.SpringBeanName;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Random;
+import static spring.SpringContextHolder.getBean;
 
 public class OptionService {
   // private OptionDao optionDAO = new OptionDaoImpl();
-  private OptionDao optionDAO = new HibernateOptionDao();
-  private CarOptionDao carOptionDAO = new CarOptionDaoImpl();
- // private CarOptionDao carOptionDAO = new HibernateCarOptionDao();
-
-
-
-
+  private OptionDao optionDao = new HibernateOptionDao();
+  //OptionDao optionDao = (OptionDao) getBean(SpringBeanName.OPTION_DAO.getName());
+ //  private CarOptionDao carOptionDao = new CarOptionDaoImpl();
+  // CarOptionDao carOptionDao = (CarOptionDao) getBean(SpringBeanName.CAR_OPTION_DAO.getName());
+  private CarOptionDao carOptionDao = new HibernateCarOptionDao();
+    public  OptionService()  { }
+    public OptionService(OptionDao optionDao) {
+        this.optionDao = optionDao;
+    }
     public Collection<Option>  getOptionsByCarId(String carId) {
-        Collection<String> optionIds = carOptionDAO.getByCarId(carId);
+        Collection<String> optionIds = carOptionDao.getByCarId(carId);
         Collection<Option> out = new ArrayList<>();
         for (String optionId : optionIds) {
-            //out.add(optionDAO.getById(optionId));//todo: 27 OptionCache
             out.add(OptionCache.getById(optionId));
-
         }
         return out;
     }
     public Collection<Option>   getOptionsByCarIdAdd(String carId) {
-        Collection<String> optionIds = carOptionDAO.getOptionsByCarIdForAdd(carId);
+        Collection<String> optionIds = carOptionDao.getOptionsByCarIdForAdd(carId);
         Collection<Option> out = new ArrayList<>();
         for (String optionId : optionIds) {
-            out.add(optionDAO.getById(optionId));
-
+            out.add(optionDao.getById(optionId));
         }
         return out;
     }
-
-//
-//    public Collection<Car> getAllCars() {
-//        Collection<Car> out = CarCache.getAllCars();
-//        for (Car c : out) {
-//            c.setOptions(optionService.getOptionsByCarId(c.getId()));
-//        }
-//        return out;
-//    }
-//
-//    public Car getById(String id) {
-//        Car out = CarCache.getById(id);
-//        out.setOptions(optionService.getOptionsByCarId(out.getId()));
-//        return out;
-//    }
-//
     public CarOption addNewOptionCar(String id, String CarId) {
         CarOption co = new CarOption(new Random().nextInt(), id, CarId);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [DMI: Random object created and used only once (DMI_RANDOM_USED_ONLY_ONCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dmi-random-object-created-and-used-only-once-dmi-random-used-only-once).
This code creates a java.util.Random object, uses it to generate one random number, and then discards the Random object. This produces mediocre quality random numbers and is inefficient. If possible, rewrite the code so that the Random object is created once and saved, and each time a new random number is required invoke a method on the existing Random object to obtain it.
If it is important that the generated Random numbers not be guessable, you *must* not create a new Random for each random number; the values are too easily guessable. You should strongly consider using a java.security.SecureRandom instead (and avoid allocating a new SecureRandom for each random number needed).